components/script/dom/element.rs

```diff
@@ -348,60 +348,50 @@ impl Element {
     }
 
     // https://drafts.csswg.org/cssom-view/#css-layout-box
-    //
-    // We'll have no content box if there's no fragment for the node, and we use
-    // bounding_content_box, for simplicity, to detect this (rather than making a more specific
-    // query to the layout thread).
     fn has_css_layout_box(&self) -> bool {
-        self.upcast::<Node>().bounding_content_box().is_some()
+        let style = self.upcast::<Node>().style();
+        style.map_or(false, |s| !s.get_box().clone_display().is_none())
     }
 
     // https://drafts.csswg.org/cssom-view/#potentially-scrollable
     fn potentially_scrollable(&self) -> bool {
-        self.has_css_layout_box() &&
-        !self.overflow_x_is_visible() &&
-        !self.overflow_y_is_visible()
+        self.has_css_layout_box() && !self.has_visible_overflow()
     }
 
     // https://drafts.csswg.org/cssom-view/#scrolling-box
     fn has_scrolling_box(&self) -> bool {
         // TODO: scrolling mechanism, such as scrollbar (We don't have scrollbar yet)
         //       self.has_scrolling_mechanism()
-        self.overflow_x_is_hidden() ||
-        self.overflow_y_is_hidden()
+        self.has_hidden_overflow()
     }
 
     fn has_overflow(&self) -> bool {
         self.ScrollHeight() > self.ClientHeight() ||
         self.ScrollWidth() > self.ClientWidth()
     }
 
-    // used value of overflow-x is "visible"
-    fn overflow_x_is_visible(&self) -> bool {
-        let window = window_from_node(self);
-        let overflow_pair = window.overflow_query(self.upcast::<Node>().to_trusted_node_address());
-        overflow_pair.x == overflow_x::computed_value::T::Visible
-    }
+    /// Computed value of overflow-x or overflow-y is "visible"
+    fn has_visible_overflow(&self) -> bool {
```

Comment:
There's duplication between `has_visible_overflow` and `has_hidden_overflow`. Ideally I'd prefer to have a more generic `fn has_some_overflow(&self, value: OverflowType)` but that would need `overflow-x` and `overflow-y` to both use the same value type I think.

Reply:
Yeah, that implies moving them outside of mako, which is not too bad now, but should definitely be a followup.

What about noting that this can be simplified after fixing #19183?